### PR TITLE
cmake: fix RISC-V toolchains

### PR DESCRIPTION
**Pull request by alalek**
- RVV options are moved to configuration scripts instead of toolchains

```
cmake ... -DCPU_BASELINE=RVV
```

Extra cmake options:
- `RISCV_RVV_SCALABLE`: ON/OFF
- redefine `CPU_RVV_FLAGS_ON` if `-march=rv64gcv` should be replaced e.g. for `-march=rv64gcv_zfh`

---

List of changes:
- `riscv64-071-gcc.toolchain.cmake` - no changes (deprecated)
- `riscv64-clang.toolchain.cmake` - dropped `v` to avoid forcing of RVV, moved out `SCALABLE` option from toolchain
- `riscv64-gcc.toolchain.cmake` - dropped `v` to avoid forcing of RVV
- `riscv64-gnu.toolchain.cmake` - dropped on favor of `riscv-gnu.toolchain.cmake`
- `riscv.toolchain.cmake` - fixed multiple issues and renamed to `riscv-gnu.toolchain.cmake`

<cut/>

```
allow_multiple_commits=1
force_builders=Custom
Xbuild_image:Custom=riscv-gcc
build_image:Custom=riscv-gcc-rvv
Xbuild_image:Custom=riscv-clang
Xbuild_image:Custom=riscv-clang-rvv
Xbuild_image:Custom=riscv-clang-rvv-128
test_modules:Custom=core,imgproc,dnn
buildworker:Custom=linux-1
test_timeout:Custom=600
build_contrib:Custom=OFF
```

### CODE REVIEW

**Comment by hanliutong** [comment]
Hello @alalek , could you please point out how can we enable RVV with this patch?

**Comment by alalek** [comment]
RVV handling becomes similar to SSE/AVX/NEON/PPC/MSA, through `CPU_BASELINE` cmake option. Updated PR's description.

File: cmake/OpenCVCompilerOptimizations.cmake [hunk 1]
```diff
@@ -375,11 +376,19 @@ elseif(PPC64LE)
...
   ocv_update(CPU_RVV_TEST_FILE "${OpenCV_SOURCE_DIR}/cmake/checks/cpu_rvv.cpp")
   ocv_update(CPU_KNOWN_OPTIMIZATIONS "RVV")
-  ocv_update(CPU_RVV_FLAGS_ON "")
-  set(CPU_DISPATCH "RVV" CACHE STRING "${HELP_CPU_DISPATCH}")
-  set(CPU_BASELINE "RVV" CACHE STRING "${HELP_CPU_BASELINE}")
+  if(RISCV_RVV_SCALABLE)
+    ocv_update(CPU_RVV_FLAGS_ON "-march=rv64gcv -DCV_RVV_SCALABLE")
+  else()
+    ocv_update(CPU_RVV_FLAGS_ON "-march=rv64gcv")
+  endif()
```
Comment (mshabunin) [inline cmake/OpenCVCompilerOptimizations.cmake:387]:
```suggestion
  ocv_update(CPU_RVV_FLAGS_ON "-march=rv64gcv")
  if(RISCV_RVV_SCALABLE)
    set(CPU_RVV_FLAGS_ON "${CPU_RVV_FLAGS_ON} -DCV_RVV_SCALABLE" CACHE INTERNAL "")
  endif()
```
This way user won't forget to add `CV_RVV_SCALABLE` definition if they set custom `CPU_RVV_FLAGS_ON` line.

Comment (alalek) [reply cmake/OpenCVCompilerOptimizations.cmake:387]: Got it.
I will check if we really need `CACHE INTERNAL` here tomorrow.

**Review by mshabunin — APPROVED** [review]

**Pull request merged**